akka-remote/src/main/scala/akka/remote/artery/compress/OutboundActorRefCompression.scala

```diff
@@ -101,6 +107,39 @@ private[remote] class OutboundCompressionTable[T](system: ActorSystem, remoteAdd
 
 }
 
+class HashScanIndexMap[T](val size: Int) {
+  private[this] val hashes: Array[Int] = Array.ofDim(size)
+  private[this] val items: Array[T] = Array.ofDim[Object](size).asInstanceOf[Array[T]]
+
+  def find(t: T): Int =
+    findItemIdx(t.hashCode(), t)
+
+  def put(t: T, i: Int): Unit = {
+    hashes(i) = t.hashCode()
+    items(i) = t
+  }
+
+  private final def findItemIdx(hashCode: Int, t: T): Int =
+    findItemIdx(0, hashCode, t)
+  @tailrec private final def findItemIdx(searchFromIndex: Int, hashCode: Int, t: T): Int =
+    if (searchFromIndex == -1) -1
+    else if (Objects.equals(items(searchFromIndex), t)) searchFromIndex
+    else findItemIdx(findHashIdx(searchFromIndex + 1, hashCode), hashCode, t)
+
+  final private def findHashIdx(searchFromIndex: Int, hashCode: Int): Int =
+    findEqIndex(hashes, searchFromIndex, hashCode)
+
+  private final def findEqIndex(hashes: Array[Int], searchFromIndex: Int, hashCode: Int): Int = {
```

Comment:
`hashes` hides the the other `val` with that name.
this `def is only used in one place, can it be inlined for clarity